helpers/services.js

```diff
@@ -233,6 +234,92 @@ function recaptchaRequiredForIp(ip) {
     return location && location.country && location.country.iso_code !== 'CN';
 }
 
+/**
+ * Constructs an InfluxDB query to get faucet data for each funnel step.
+ *
+ * @param {string} dateFrom RFC3339 UTC
+ * @param {string} dateTo RFC3339 UTC
+ * @returns {string} InfluxDB query
+ */
+const buildFaucetInfluxQuery = (dateFrom, dateTo) =>
+    checkpoints
+        .reduce(
+            (acc, cur) => [
+                ...acc,
+                encodeURIComponent(
+                    `SELECT COUNT("step") AS "${
+                        cur.symbol
+                    }" FROM "overseer"."autogen"."signup" WHERE time >= '${dateFrom}' AND time <= '${dateTo}' AND "step"='${
+                        cur.symbol
+                    }'; `
+                ),
+            ],
+            []
+        )
+        .join('');
+
+/**
+ * Retrieves signup data from InfluxDB and formats for use in the dashboard.
+ * Returns an array where each element includes:
+ *  - a human-readable label `human`,
+ *  - `symbol`, the db column name that matches the frontend step constants,
+ *  - `count`
+ *  - `percent` calculated by dividing the total signup events for this period by this event
+ *
+ * @param {Date} dateFrom RFC3339 UTC
+ * @param {Date} dateTo RFC3339 UTC
+ * @return {Array}
+ */
+async function getOverseerStats(dateFrom, dateTo) {
+    const response = await (await fetch(
+        `${process.env.INFLUXDB_URL}/query?db=overseer`,
+        {
+            method: 'POST',
+            headers: new Headers({
+                'Content-Type': 'application/x-www-form-urlencoded',
+            }),
+            body: `q=${buildFaucetInfluxQuery(dateFrom, dateTo)}`,
+        }
+    )).json();
+    if (!response.results) {
+        throw new Error('influxdb query error');
+    }
+
+    let mappedResults;
+    try {
+        mappedResults = response.results.reduce(
+            (acc, cur) => ({
+                ...acc,
+                [cur.series[0].columns[1]]: cur.series[0].values[0][1] || null,
```

Comment:
it seems that `cur` object does not always have the `series` key...

e.g. adding console logs like this:
```
    try {
        mappedResults = response.results.reduce(
            (acc, cur) => {
                console.log('cur:', cur)
                return ({
                    ...acc,
                    [cur.series[0].columns[1]]: cur.series[0].values[0][1] || null,
                })
            },
            {}
        );
    } catch (error) {
        console.log('error:', error)
        throw new Error('influxdb data error');
    }
```

```
cur: { statement_id: 3 }
error: TypeError: Cannot read property '0' of undefined
    at response.results.reduce (/home/iain/faucet/helpers/services.js:295:28)
    at Array.reduce (<anonymous>)
    at Object.getOverseerStats (/home/iain/faucet/helpers/services.js:290:42)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```